src/explorer/AzureAccountTreeItem.ts

```diff
@@ -2,17 +2,203 @@
+      if (GeneralUtils.failed(session) || !AzureAuth.isReady(this.sessionProvider)) {
+        return [
+          new GenericTreeItem(this, {
+            label: UiStrings.ErrorAuthenticating,
+            contextValue: "azureCommand",
+            id: APIMAccountCommandId.accountError,
+            iconPath: new vscode.ThemeIcon("error"),
+          }),
+        ];
+      }
+  
+      const subscriptions = await AzureSubscriptionHelper.getSubscriptions(this.sessionProvider, SelectionType.Filtered);
+      if (GeneralUtils.failed(subscriptions)) {
+        return [
+          new GenericTreeItem(this, {
+            label: UiStrings.ErrorLoadingSubscriptions,
+            contextValue: "azureCommand",
+            id: APIMAccountCommandId.accountError,
+            iconPath: new vscode.ThemeIcon("error"),
+            description: subscriptions.error,
+          }),
+        ];
+      }
+  
+      if (subscriptions.result.length === 0) {
+        return [
+          new GenericTreeItem(this, {
+            label: UiStrings.NoSubscriptionsFound,
+            contextValue: "azureCommand",
+            id: APIMAccountCommandId.accountError,
+            iconPath: new vscode.ThemeIcon("info"),
+          }),
+        ];
+      }
+  
+      // We've confirmed above that the provider is ready.
+      const readySessionProvider: ReadyAzureSessionProvider = this.sessionProvider;
+  
+      this.subscriptionTreeItems = await Promise.all(
+        subscriptions.result.map(async (subscription: any) => {
+          const existingTreeItem: AzExtTreeItem | undefined = existingSubscriptionTreeItems.find(
+            (ti) => ti.id === subscription.subscriptionId,
+          );
+          if (existingTreeItem) {
+            return existingTreeItem;
+          } else {
+            const subscriptionContext = getSubscriptionContext(
+              readySessionProvider,
+              session.result,
+              subscription,
+            );
+            return await createSubscriptionTreeItem(this, subscriptionContext);
+          }
+        }),
+      );
+  
+      return this.subscriptionTreeItems!;
+    }
+  }
+  
+  function getSubscriptionContext(
+    sessionProvider: ReadyAzureSessionProvider,
+    session: vscode.AuthenticationSession,
+    subscription: Subscription,
+  ): ISubscriptionContext {
+    const credentials = AzureAuth.getCredential(sessionProvider);
+    const environment = AzureAuth.getEnvironment();
+  
+    return {
+      credentials,
+      createCredentialsForScopes: ()=> { return Promise.resolve(credentials); },
+      subscriptionDisplayName: subscription.displayName || "",
+      subscriptionId: subscription.subscriptionId || "",
+      subscriptionPath: `/subscriptions/${subscription.subscriptionId}`,
+      tenantId: subscription.tenantId || "",
+      userId: session.account.id,
+      environment,
+      isCustomCloud: environment.name === "AzureCustomCloud",
```

Comment:
potential issue here. We only support Azure custom cloud now? Please check with Chaoyi. And how about if there is ask to support other clouds?

Reply:
I saw US Government cloud has APIM service (other environment to be confirm), so we have the requirement to support other cloud. Wenyu is investigating the effort while I confirming the environment list as well as priority.

Reply:
according to the azure-account base subtreeitembase class, also defined `iscustomCloud: filter.session.environment.name === 'AzureCustomCloud'`
https://github.com/microsoft/vscode-azuretools/blob/c71004cdb205e3cf512f7e792472c5f768468f28/azure/src/tree/AzureAccountTreeItemBase.ts#L154